CONTRIBUTING.md

```diff
@@ -1,59 +1,235 @@
+| `feature/*`      | NO         | `dev`              | Cutting-edge features (**unstable**). These branches are used for any maintenance features / active development.                                                                                              |
+| `release-vX.Y.Z` | NO         | `dev`              | A temporary release branch that follows the [semver](http://semver.org/) versioning. This is what is sent to UAT.<br/>A pull request is required to merge code into any `release-vX.Y.Z` branch.              |
+| `hotfix/*`       | NO         | `main`             | These are bug fixes against production.<br/>This is used because develop might have moved on from the last published state.<br/>Remember to merge this back into develop and any release branches.            |
 
-To send us a pull request, please:
+## Use Cases
 
-1. Fork the repository.
-2. Modify the source; please focus on the specific change you are contributing. If you also reformat all the code, it will be hard for us to focus on your change.
-3. Ensure local tests pass.
-4. Commit to your fork using clear commit messages.
-5. Send us a pull request, answering any default questions in the pull request interface.
-6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
+### Develop a new feature
 
-GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
-[creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+![Feature branch](images/feature-branch.png)
 
+1. Make sure your `dev` branch is up-to-date
 
-## Finding contributions to work on
-Looking at the existing issues is a great way to find something to contribute on. As our projects, by default, use the default GitHub issue labels (enhancement/bug/duplicate/help wanted/invalid/question/wontfix), looking at any 'help wanted' issues is a great place to start.
+2. Create a feature branch based off of `dev`
 
+   ```
+   $ git checkout dev
+   $ git checkout -b feature-new-documentation
+   $ git push --set-upstream feature-new-documentation
+   ```
 
-## Code of Conduct
-This project has adopted the [Amazon Open Source Code of Conduct](https://aws.github.io/code-of-conduct).
-For more information see the [Code of Conduct FAQ](https://aws.github.io/code-of-conduct-faq) or contact
-opensource-codeofconduct@amazon.com with any additional questions or comments.
+3. Develop the code for the new feature and commit. Push your changes often. This
+   allows others to see your changes and suggest improvements/changes as well as
+   provides a safety net should your hard drive crash.
 
+    ```
+    $ ... make changes
+    $ git add -A .
+    $ git commit -m "Add new documentation files"
+    $ ... make more changes
+    $ git add -A .
+    $ git commit -m "Fix some spelling errors"
+    $ git push
```

Comment:
I also generally squash multiple commits before putting out the final PR. It looks like GitHub has some automation about that built in so maybe it's not critical these days but something to think about.